[Refactor] Domain 타입 리팩토링 및 Preview DEBUG 용 타입 생성

Marcorable opened the pull request:
## 📌 배경

merged #112 

## 내용
1. 리팩토링 Domain
   > #112 에서 반영하지 못한 오류 및 변경 사항 적용
2. Preview DEBUG 용 타입 생성
   > 모듈이 달라서 Preview에서 사용 못하는 타입 Mock? Fake? (일단 Mock 으로 명명 ㅠㅠ) 데이터 생성

Marcorable committed:
refactor: Tag 및 Article 수정

- ArticleData. TagData dummy 데이터 생성
- ArticleModelProtocol 및 ArticleModel 수정
- Tag 및 TagData 수정

Marcorable committed:
feat: Preview를 위한 DEBUG 타입 생성

- Domain 타입
- UserInterface 타입

joseph704 reviewed (approved):
고생하셨습니다!

joseph704 commented on Targets/UserInterface/Sources/ContentView.swift:
```diff
@@ -27,3 +27,57 @@ public struct ContentView: View {
+    var isPublic: Bool
+    var tagIds: [Int]
+    var created: String
+}
+
+struct MockLoginUser: LoginUser {
+    var nickName: String = ""
+    var accessToken: String?
+}
+
+final class MockArticleModel: ArticleModelProtocol {
+    @Published var items: [DomainInterface.Article] = []
+    var itemsPublisher: Published<[DomainInterface.Article]>.Publisher { $items }
+
+    func fetch() { }
+    func create(_ item: DomainInterface.Article) { }
+    func read(_ item: DomainInterface.Article) { }
+    func update(_ item: DomainInterface.Article) async { }
+    func remove(_ item: DomainInterface.Article) { }
+}
+
+final class MockTagModel: TagModelProtocol {
+    @Published var items: [DomainInterface.Tag] = []
+    var itemsPublisher: Published<[DomainInterface.Tag]>.Publisher { $items }
+    
+    func fetch() { }
+    func create(_ item: DomainInterface.Tag) { }
+    func read(_ item: DomainInterface.Tag) { }
+    func update(_ item: DomainInterface.Tag) { }
+    func remove(_ item: DomainInterface.Tag) { }
+}
+
+final class MockLoginModel: LoginModelProtocol {
+    @Published var userData: LoginUser = MockLoginUser()
+    var userDataPublisher: Published<DomainInterface.LoginUser>.Publisher { $userData }
+    
+    func checkKakaoLogin() async -> Bool { false }
+}
+
+#endif
```
import DomainInterface가 빠지것 같습니다. 해당 부분에 대해 컴파일 오류가 나는데 확인 부탁드립니다! 

1. MockLoginModel에서 LoginModelProtocol 미준수 
2. MockArticle에서 Article 미준수 
3. HomeViewMdoel.swift 110번째 줄 id파라미터 데이터 타입 Int로 변경해야함

Marcorable replied:
@joseph704 

전체 타겟으로 빌드하니 못찾아내내요 ㅠㅠ 역시 엑코가 엑코하는..

[commit 50e4368](https://github.com/depromeet/ticlemoa-iOS/pull/120/commits/50e4368e49370e04578cf525f84160c3f8fb2c9b) 추가했습니다
확인 감사합니다!

Marcorable committed:
fix: 중복 네이밍 수정 및 누락 변경사항 반영

- Domain Article 과 UI 모듈 Article 네이밍 충돌
   -> UI 모듈 Article은 임시 타입으로 변형 또는 제거 예정
- id 타입은 UUID에서 Int로 변경됨 (누락사항)

kipsong133 reviewed (approved):
Publisher 를 잘 사용해보지 않아서 어색한데, 적용해주셔서 감사해요~!

sustainable-git reviewed (approved):
수고하셨습니다!!!
